source/common/http/conn_pool_grid.cc

```diff
@@ -102,6 +103,11 @@ bool ConnectivityGrid::WrapperCallbacks::shouldAttemptSecondHttp3Connection() {
   if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.http3_happy_eyeballs")) {
     return false;
   }
+  // Make sure the original caller hasn't been called. This shouldn't be
+  // possible but is a helpful sanity check.
```

Comment:
Let's IS_ENVOY_BUG if we think this is impossible. (Trust, but verify :>)

Reply:
removed and tested what happens when it happens instead